core/drivers/clk/clk-stm32mp15.c

```diff
@@ -1350,6 +1382,264 @@ static TEE_Result stm32mp1_clk_fdt_init(const void *fdt, int node)
+ *
+ * stm32mp1_clk first cells follow stm32mp1_clk_gate[] ordering.
+ * stm32mp1_clk last cells follow stm32mp1_clk_on[] ordering.
+ */
+static struct clk stm32mp1_clk[ARRAY_SIZE(stm32mp1_clk_gate) +
+			       ARRAY_SIZE(stm32mp1_clk_on)];
+
+#define CLK_ON_INDEX_OFFSET	((int)ARRAY_SIZE(stm32mp1_clk_gate))
+
+static bool clk_is_gate(struct clk *clk)
+{
+	int clk_index = clk - stm32mp1_clk;
+
+	assert(clk_index >= 0 && clk_index < (int)ARRAY_SIZE(stm32mp1_clk));
+	return clk_index < CLK_ON_INDEX_OFFSET;
+}
+
+static unsigned long clk_to_clock_id(struct clk *clk)
+{
+	int gate_index = clk - stm32mp1_clk;
+	int on_index = gate_index - CLK_ON_INDEX_OFFSET;
+
+	if (clk_is_gate(clk))
+		return stm32mp1_clk_gate[gate_index].clock_id;
+
+	return stm32mp1_clk_on[on_index];
+}
+
+static const struct stm32mp1_clk_gate *clk_to_gate_ref(struct clk *clk)
+{
+	int gate_index = clk - stm32mp1_clk;
+
+	assert(clk_is_gate(clk));
+
+	return stm32mp1_clk_gate + gate_index;
+}
+
+static struct clk *clock_id_to_clk(unsigned long clock_id)
+{
+	int gate_index = clock_id_to_gate_index(clock_id);
+	int on_index = clock_id_to_always_on_index(clock_id);
+
+	if (gate_index >= 0)
+		return stm32mp1_clk + gate_index;
+
+	if (on_index >= 0)
+		return stm32mp1_clk + CLK_ON_INDEX_OFFSET + on_index;
+
+	return NULL;
+}
+
+/* Only store names of clocks that are found from their name */
+struct clk_name {
+	unsigned int clock_id;
+	const char *name;
+};
+
+#define CLOCK_NAME(_binding, _name) \
+	{ .clock_id = (_binding), .name = (_name) }
+
+const struct clk_name exposed_clk_name[] = {
+#if CFG_TEE_CORE_LOG_LEVEL > TRACE_DEBUG
+	/* Clocks used by platform drivers not yet probed from DT */
+	CLOCK_NAME(CK_DBG, "dbg"),
+	CLOCK_NAME(CK_MCU, "mcu"),
+	CLOCK_NAME(RTCAPB, "rtcapb"),
+	CLOCK_NAME(BKPSRAM, "bkpsram"),
+	CLOCK_NAME(RTC, "rtc"),
+	CLOCK_NAME(CRYP1, "crpy1"),
+	CLOCK_NAME(SYSCFG, "syscfg"),
+	CLOCK_NAME(GPIOA, "gpioa"),
+	CLOCK_NAME(GPIOB, "gpiob"),
+	CLOCK_NAME(GPIOC, "gpioc"),
+	CLOCK_NAME(GPIOD, "gpiod"),
+	CLOCK_NAME(GPIOE, "gpioe"),
+	CLOCK_NAME(GPIOF, "gpiof"),
+	CLOCK_NAME(GPIOG, "gpiog"),
+	CLOCK_NAME(GPIOH, "gpioh"),
+	CLOCK_NAME(GPIOI, "gpioi"),
+	CLOCK_NAME(GPIOJ, "gpioj"),
+	CLOCK_NAME(GPIOK, "gpiok"),
+	CLOCK_NAME(GPIOZ, "gpioz"),
+	/* Clock exposed by SCMI. SCMI clock fmro DT bindings to come... */
+	CLOCK_NAME(CK_HSE, "hse"),
+	CLOCK_NAME(CK_HSI, "hsi"),
+	CLOCK_NAME(CK_CSI, "csi"),
+	CLOCK_NAME(CK_LSE, "lse"),
+	CLOCK_NAME(CK_LSI, "lsi"),
+	CLOCK_NAME(PLL2_Q, "pll2q"),
+	CLOCK_NAME(PLL2_R, "pll2r"),
+	CLOCK_NAME(PLL3_Q, "pll3q"),
+	CLOCK_NAME(PLL3_R, "pll3r"),
+	CLOCK_NAME(CRYP1, "cryp1"),
+	CLOCK_NAME(HASH1, "hash1"),
+	CLOCK_NAME(I2C4_K, "i2c4"),
+	CLOCK_NAME(I2C6_K, "i2c6"),
+	CLOCK_NAME(IWDG1, "iwdg"),
+	CLOCK_NAME(RNG1_K, "rng1"),
+	CLOCK_NAME(SPI6_K, "spi6"),
+	CLOCK_NAME(USART1_K, "usart1"),
+	CLOCK_NAME(CK_MCU, "mcu"),
+#endif
+	/* Last element used as default name reference */
+	CLOCK_NAME(UINT8_MAX, "n.a")
```

Comment:
actually not needed, i'll remove and change `clk_op_get_name()` to return `NULL`.